NameLayerMain/src/vg/civcraft/mc/namelayer/command/commands/ListPermissions.java

```diff
@@ -33,34 +33,34 @@ public boolean execute(CommandSender sender, String[] args) {
 			return true;
 		}
 		Player p = (Player) sender;
-		Group g = gm.getGroup(args[0]);
+		Group g = GroupManager.getGroup(args[0]);
 		if (groupIsNull(sender, args[0], g)) {
 			return true;
 		}
 		UUID uuid = NameAPI.getUUID(p.getName());
-		PlayerType playerType = g.getPlayerType(uuid);
-		if (playerType == null){
-			p.sendMessage(ChatColor.RED + "You do not have access to this group.");
-			return true;
-		}
-		String perms = null;
-		GroupPermission gPerm = gm.getPermissionforGroup(g);
+		StringBuilder sb = new StringBuilder();
 		if(args.length > 1){
 			if (!gm.hasAccess(g, uuid, PermissionType.getPermission("LIST_PERMS"))){
 					p.sendMessage(ChatColor.RED + "You do not have permission in this group to run this command.");
 					return true;
 			}
-			PlayerType check = PlayerType.getPlayerType(args[1]);
-			if (check == null){
-				PlayerType.displayPlayerTypes(p);
-				return true;
+			for(PlayerType pType : g.getPlayerTypeHandler().getAllTypes()) {
+					sb.append("[" + pType.getName() +"-PERMS]: ");
```

Comment:
This was copied from the /nldig function yeah? I think it would look a bit weird here having [<name>-PERMS]:. Perhaps just long form it -- `pType.getName() + " Permissions: "` ? Same a few lines after this.